src/dir.cr

```diff
@@ -305,6 +309,27 @@ class Dir
       self
     end
   end
+
+  private struct ChildIterator
+    include Iterator(String)
+
+    def initialize(@dir : Dir)
+    end
+
+    def next
+      excluded = {".", ".."}
+      entry = nil
+      loop do
+        break unless excluded.includes?(entry = @dir.read)
```

Comment:
Can I suggest another one?
```cr
excluded = {".", ".."}
while entry = @dir.read
  return entry unless excluded.includes? entry
end
stop
```
Nevermind you don't have to include that one, just wanted to write it somewhere..

Edit: oh in fact you implemented it that way too, perfect then 😆